### PR TITLE
params: add WithBootstrappers option

### LINKED ISSUE #785: params: BootstrapPeers type and WithBootstrapPeers option for the node
Currently, there is a `params.Network` type that is registered on the Node that defaults to `params.Network` and has an option/modifier to alter it. Similar to this, we should have `params.Bootstrappers` type with defaults taken from `params.BootstrappersFor`(as there are right now) and with the option to modify those on the Node.

### PR DESCRIPTION
**Pull request by vgonkivs**
Resolves #785

### CODE REVIEW

File: params/network.go [hunk 1]
```diff
@@ -16,6 +18,9 @@ const (
 // Network is a type definition for DA network run by Celestia Node.
 type Network string
 
+// BootstrapPeers is a type definition for nodes that will be used as bootstrappers
+type BootstrapPeers []*peer.AddrInfo
```
Comment (Wondertan) [inline params/network.go:22]: We need to be consistent with namings. Our func in the pkg is called BootstrappersFor while the type is BootstrapPeer.

Comment (Wondertan) [reply params/network.go:22]: Also, now I think we should keep peer.AddrInfo as a value(not pointer) to be consistent with DHT and they libp2p uses them

File: node/tests/sync_test.go [hunk 2]
```diff
@@ -281,3 +282,26 @@ func TestSyncLightWithTrustedPeers(t *testing.T) {
 
 	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 50))
 }
+
+func TestUseBridgeNodeAsBootstraper(t *testing.T) {
```
Comment (Wondertan) [inline node/tests/sync_test.go:286]: Please make another file `p2p.go` and move it there.

**Review by Wondertan — COMMENTED** [review]
That was quick, awesome! Thanks.

File: node/node.go [hunk 3]
```diff
@@ -36,9 +36,10 @@ var log = logging.Logger("node")
 // * Light
 // * Full
 type Node struct {
-	Type    Type
-	Network params.Network
-	Config  *Config
+	Type           Type
+	Network        params.Network
+	BootstrapPeers params.Bootstrappers
```
Comment (Wondertan) [inline node/node.go:41]: Bootstrappers

File: params/bootstrap.go [hunk 4]
```diff
@@ -9,17 +9,17 @@ import (
...
 		return nil, err
 	}
 
 	return parseAddrInfos(bs)
 }
 
-// BootstrappersFor reports multiaddresses of bootstrap peers for a given network.
-func BootstrappersFor(net Network) ([]string, error) {
+// bootstrappersFor reports multiaddresses of bootstrap peers for a given network.
+func bootstrappersFor(net Network) ([]string, error) {
```
Comment (Wondertan) [inline params/bootstrap.go:22]: If me make this private, we can omit Infos from BootstrappersInfoFor to make shorter. Also, BootstrappersInfoFor returns the type Bootstrappers and not the info for the type bootstrappers

**Review by Wondertan — APPROVED** [review]
Last bits and lfg

**Review by Wondertan — APPROVED** [review]

File: node/services/config.go [hunk 5]
```diff
@@ -30,26 +30,23 @@ func DefaultConfig() Config {
...
-		cfg.TrustedPeers, err = params.BootstrappersFor(net)
-		if err != nil {
-			return
-		}
+		log.Infof("No trusted peers in config, initializing with default bootstrappers as trusted peers")
+		return bpeers, nil
 	}
 
-	infos = make([]*peer.AddrInfo, len(cfg.TrustedPeers))
+	infos = make([]peer.AddrInfo, len(cfg.TrustedPeers))
```
Comment (renaynay) [inline node/services/config.go:39]: Why this change?

Comment (vgonkivs) [reply node/services/config.go:39]: to be consistent wit libp2p. It requires `peer.AddrInfo` in `dht.BootstrapPeers()`
